### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by sanchopanca**
This also should fix issues  that are connected to Settings class.

Tested in Sublime 2 & 3 under Linux x64

### CODE REVIEW

**Comment by zirho** [comment]
we need this PR

**Comment by sanchopanca** [comment]
Yes, @condemil, any feedback?

**Comment by rudiedirkx** [comment]
How do I test a package's dev version, like `sanchopanca/Gist:master`?

**Comment by rudiedirkx** [comment]
And could you make a diff without your editor's auto corrections?, because this isn't very readable for seeing what has actually changed.

**Comment by zirho** [comment]
check this one out 
http://www.macdrifter.com/2012/08/install-sublime-packages-from-github.html
I installed it from his repo

**Comment by sanchopanca** [comment]
@rudiedirkx These changes aren't no my editor's auto corrections. They are my changes.

File: gist.py [hunk 1]
```diff
@@ -216,8 +214,8 @@ def on_gist_filename(filename):
                 sublime.status_message("%s Gist: %s" % (self.mode(), gist_html_url))
 
                 if gistify:
-                    gistify_view(self.view, gist, list(gist['files'].keys())[0])
-                # else:
+                    helpers.gistify_view(self.view, gist, list(gist['files'].keys())[0])
+                    # else:
                     # open_gist(gist['url'])
```
Comment (rudiedirkx) [inline gist.py]: Really @sanchopanca ? You changed this indentation manually? Looks like editor autocorrection. Sorry.

**Comment by sanchopanca** [comment]
@rudiedirkx I wrote this code in PyCharm and it doesn't auto-correct such comments, I just [checked](http://imgur.com/gbb0xIA). I don't know why I changed this line, to be honest. This was the commit where I tried to fix flake8's errors. Maybe I should remove these comments completely.

**Pull request merged**